src/library_noderawfs.js

```diff
@@ -96,16 +96,33 @@ mergeInto(LibraryManager.library, {
       var newMode = NODEFS.getMode(pathTruncated);
       var fd = suggestFD != null ? suggestFD : FS.nextfd(nfd);
       var node = { id: st.ino, mode: newMode, node_ops: NODERAWFS, path: path }
-      var stream = { fd: fd, nfd: nfd, position: 0, path: path, flags: flags, node: node, seekable: true };
+      var stream = FS.createStream({ fd: fd, nfd: nfd, position: 0, path: path, flags: flags, node: node, seekable: true });
       FS.streams[fd] = stream;
       return stream;
     },
+    createStream: function(stream, fd_start, fd_end){
+      // Call the original FS.createStream
+      var rtn = VFS.createStream(stream, fd_start, fd_end);
+      if (typeof rtn.shared.refcnt == "undefined") {
+        rtn.shared.refcnf = 0;
+      } else {
+        rtn.shared.refcnf++;
+      }
+      return rtn;
+   },
+   closeStream: function(fd) {
+     if(FS.streams[fd].shared){
+       FS.streams[fd].shared.refcnt--;
+     }
+     VFS.closeStream(fd);
+   },
     close: function(stream) {
-      if (!stream.stream_ops) {
-        // this stream is created by in-memory filesystem
+      FS.closeStream(stream.fd);
+      if (!stream.stream_ops && (!stream.shared || stream.shared.refcnt === 0)) {
```

Comment:
Is it possible to have a stream without `.shared` field?

Reply:
Apparently. See `test_pthread_dlsym` failure in [this job](https://app.circleci.com/pipelines/github/emscripten-core/emscripten/20026/workflows/69db45fb-7451-42d5-9505-b337ae2bbd15/jobs/517042). These guards fix it. I am not sure how this code path is hit with a stream that doesn't have a `shared`. Somehow a stream was created without calling `createStream`? But then it was destroyed with noderawfs `closeStream`? and what does it have to do with pthreads and dynamic linking? I don't understand, but hopefully the tests will pass anyways.

Reply:
Looks like maybe `createStandardStreams` above needs to be converted to call createStream?

Reply:
Ah yeah that would make sense. Like this?
```js
FS.streams[0] = FS.createStream({ fd: 0, nfd: 0, position: 0, path: '', flags: 0, tty: true, seekable: false });
for (var i = 1; i < 3; i++) {
  FS.streams[i] = FS.createStream({ fd: i, nfd: i, position: 0, path: '', flags: 577, tty: true, seekable: false });
}
```

Reply:
That does the trick.

Reply:
Yes, I wonder if once you change that we can be sure that all streams have shared field?

Reply:
I suspect that it is at least true in the test suite now.